### PR TITLE
[opt]function use sse to opt match_ipv6_subnet

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

test
```
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_matchIPv6SubnetSSE          1.89 ns         1.89 ns   1000000000
BM_matchIPv6SubnetNative       4.99 ns         4.99 ns    561455254
```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

File: be/src/vec/runtime/ip_address_cidr.h [hunk 1]
```diff
@@ -73,6 +73,29 @@ bool match_ipv4_subnet(uint32_t addr, uint32_t cidr_addr, uint8_t prefix) {
...
+    if (mask) {
+        const auto offset = std::countl_zero(mask);
+        if (prefix / 8 != offset) {
+            return prefix / 8 < offset;
+        }
+        auto cmpmask = ~(0xff >> (prefix % 8));
+        return (addr[IPV6_BINARY_LENGTH - 1 - offset] & cmpmask) ==
+               (cidr_addr[IPV6_BINARY_LENGTH - 1 - offset] & cmpmask);
+    } else {
+        // All the bytes are equal.
```
Comment (HappenLee) [inline be/src/vec/runtime/ip_address_cidr.h:93]: why keep the unless else

Comment (Mryange) [reply be/src/vec/runtime/ip_address_cidr.h:93]: Just for the comment

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**